### ISSUE TITLE
New rule to prevent to use `expect(query*(...)).toBeDefined()`

### ISSUE THREAD

**Issue by retyui** [testing-library/eslint-plugin-testing-library#806]
### Name for new rule

your-name

### Description of the new rule

False Positives check, that need to be forbidden:

```tsx
query*(...) // null | Elemnt
expect(null).toBeDefined(); // true
```

### Testing Library feature

`.query*(...)`

### Testing Library framework(s)

dom 7 react-native

### What category of rule is this?

Suggests an alternate way of doing something

### Code examples

```tsx
query*(...) // null | Elemnt
expect(null).toBeDefined(); // true
expect(query*(...)).toBeDefined(); // bad
```

### Do you want to submit a pull request to make the new rule?

No

**Comment by Belco90** [comment]
Hi @retyui. You can achieve this using [the `eslint-plugin-jest/no-restricted-matchers` rule](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-restricted-matchers.md).
